core/src/main/scala/org/apache/spark/scheduler/Task.scala

```diff
@@ -206,6 +210,11 @@ private[spark] object Task {
       dataOut.writeLong(timestamp)
     }
 
+    // Write the task properties separately so it is available before full task deserialization.
+    val propBytes = Utils.serialize(task.localProperties)
```

Comment:
Just curious, why not `serializer.serialize(..)`? This is fine, but just wondeirng.